tools/serve/serve.py

```diff
@@ -922,17 +922,22 @@ def handle_signal(signum, frame):
             signal.signal(signal.SIGTERM, handle_signal)
             signal.signal(signal.SIGINT, handle_signal)
 
-            while (all(item.is_alive() for item in iter_procs(servers)) and
+            while (all(subproc.is_alive() for subproc in iter_procs(servers)) and
                    not received_signal.is_set()):
-                for item in iter_procs(servers):
-                    item.join(1)
-            exited = [item for item in iter_procs(servers) if not item.is_alive()]
-            subject = "subprocess" if len(exited) == 1 else "subprocesses"
-
-            logger.info("%s %s exited:" % (len(exited), subject))
-
-            for item in iter_procs(servers):
-                logger.info("Status of %s:\t%s" % (item.name, "running" if item.is_alive() else "not running"))
+                for subproc in iter_procs(servers):
+                    subproc.join(1)
+
+            failed_subproc = 0
+            for subproc in iter_procs(servers):
+                if subproc.is_alive():
+                    logger.info('Status of subprocess "%s": running' % subproc.name)
```

Comment:
Shouldn't we kill the subprocess here? Otherwise it'd become a daemon right?

Reply:
As far as I can test, they don't become daemons. Not sure now how they get killed, but they do (at least when ending "wpt serve" via Ctrl+C)
Do you think it's better (or more clear) to send a kill signal here?

Reply:
Ctrl-C in an interactive shell not only sends SIGINT to the parent process, but also to *all* subprocesses in the process group, so effectively each subprocess gets terminated independently. If you call `kill -INT`, you should see the subprocesses become daemons.

We can address this in another PR though. This PR on its own arguably fixes a bug (the exit code shouldn't be 0), whereas the change of signal handling is a nice-to-have improvement.

Reply:
Good point about Ctrl-C sending SIGINT to all process from an iterative shell. However, I tried now to run ```kill -INT``` over the main pid of "wpt serve" and I still see all the subprocess ending with the main wpt (no daemons left). I'm a bit puzzled, not sure about what causes them to be closed.

Reply:
For the record, I've found out how these subprocesses are terminated. They all have `daemon=True`: https://docs.python.org/2/library/multiprocessing.html#multiprocessing.Process.daemon so they will be closed by Python runtime when the main process exits.